src/api/Shop.ts

```diff
@@ -148,59 +163,101 @@ const InitShop = (u: types.FirebaseUser, id: number): types.Shop => {
+						console.error("RequestColorwayFrames recieved SizeIDOutsideRecommendedRange");
 						reject(error);
 						break;
 					default:
 						if (error instanceof Error) {
+							console.error("RequestColorwayFrames recieved unexpected error", error);
 							reject(createUIError(L.SomethingWentWrong, error));
 						} else {
 							// UIError
+							console.error("RequestColorwayFrames recieved ui error", error);
 							reject(createUIError(L.SomethingWentWrong, new Error(error.error)));
 						}
 				}
 			});
 		});
 	};
 
-	const TryOn = async (ColorwaySizeAssetSKU: string): Promise<TryOnFrames> => {
+	const TryOn = (colorwaySizeAssetSKU: string): Promise<types.TryOnFrames | Promise<types.TryOnFrames>> => {
+		let styleIDCache: number;
 		return new Promise((resolve, reject) => {
-			GetStyles().then((styles) => {
-				const colorwaySizeAssetID = LookupColorwaySizeAssetIDBySKU(ColorwaySizeAssetSKU, styles);
-				if (colorwaySizeAssetID) {
-					console.info("requesting new colorway frames");
-					RequestColorwaySizeAssetFrames(colorwaySizeAssetID).then(() => {
-						// listen for changes in firebase
-						console.info("waiting for rendered colorway frames");
-						AwaitColorwaySizeAssetFrames(ColorwaySizeAssetSKU).then((frames) => {
+			GetColorwaySizeAssetFrames(colorwaySizeAssetSKU).then((frames) => {
+				resolve(frames as types.TryOnFrames);
+			}).catch((error: Error) => {
+				if (error != types.NoFramesFound) {
+					console.error("error is not NoFramesFound", error);
+					reject(error);
+					return;
+				}
+				resolve(new Promise((resolve, reject) => {
+					// lookup current sku to get style_id
+					GetColorwaySizeAssets(null, [colorwaySizeAssetSKU])
+						.then((colorwaySizeAssets: Map<number, types.FirebaseColorwaySizeAsset>) => {
+							if (!colorwaySizeAssets?.size) {
+								console.error("no colorway size assets found");
+								return Promise.reject(types.NoColorwaySizeAssetsFound);
+							}
+							const colorwaySizeAsset = colorwaySizeAssets.values().next().value
+							styleIDCache = colorwaySizeAsset.style_id;
+							return RequestColorwaySizeAssetFrames(colorwaySizeAsset.id);
+						}).then(() => {
+							// listen for changes in firebase
+							return AwaitColorwaySizeAssetFrames(colorwaySizeAssetSKU);
+						}).then((frames) => {
 							resolve(frames);
-						}).catch((error: UIError | Error) => {
-							console.error("error awaiting colorway frames", error)
-							reject(error);
+						}).catch((error: Error | ErrorOutsideRecommendedSizes) => {
+							console.error("error requesting colorway frames", error);
+							if (!(error as ErrorOutsideRecommendedSizes).recommended_size_id) {
+								reject(error);
+								return;
+							}
+
+							GetStyles([styleIDCache]).then((styles: Map<number, types.FirebaseStyle>) => {
+								let errorOutsideRecommended = error as ErrorOutsideRecommendedSizes;
+								let recommendedSizeResponse: types.RecommendedAvaliableSizes = {
```

Comment:
RecommendedAvaliableSizes has a typo

Reply:
renamed prefixes from firebase to firestore
renamed all `avaliable` to `available`